internal/controllers/machine/machine_controller_status.go

```diff
@@ -213,8 +218,49 @@ func setInfrastructureReadyCondition(_ context.Context, machine *clusterv1.Machi
 	})
 }
 
-func setNodeHealthyAndReadyConditions(ctx context.Context, machine *clusterv1.Machine, node *corev1.Node) {
-	// TODO: handle disconnected clusters when the new ClusterCache is merged
+func setNodeHealthyAndReadyConditions(ctx context.Context, machine *clusterv1.Machine, node *corev1.Node, nodeGetErr error, lastProbeSuccessTime time.Time, remoteConditionsGracePeriod time.Duration) {
+	if time.Since(lastProbeSuccessTime) > remoteConditionsGracePeriod {
+		v1beta2conditions.Set(machine, metav1.Condition{
+			Type:    clusterv1.MachineNodeReadyV1Beta2Condition,
+			Status:  metav1.ConditionUnknown,
+			Reason:  clusterv1.MachineNodeRemoteConnectionFailedV1Beta2Reason,
+			Message: fmt.Sprintf("Remote connection down since %s", lastProbeSuccessTime.Format(time.RFC3339)),
+		})
+
+		v1beta2conditions.Set(machine, metav1.Condition{
+			Type:    clusterv1.MachineNodeHealthyV1Beta2Condition,
+			Status:  metav1.ConditionUnknown,
+			Reason:  clusterv1.MachineNodeRemoteConnectionFailedV1Beta2Reason,
+			Message: fmt.Sprintf("Remote connection down since %s", lastProbeSuccessTime.Format(time.RFC3339)),
+		})
+		return
+	}
+
+	if nodeGetErr != nil {
+		if errors.Is(nodeGetErr, clustercache.ErrClusterNotConnected) {
```

Comment:
Should we set the condition to unknown if it does not exist yet? (to ensure the condition always exists)

Reply:
Done